libkernelflinger/android.c

```diff
@@ -1161,7 +1176,11 @@ static EFI_STATUS setup_command_line(
                 char *vb_cmdline;
                 vb_cmdline = get_vb_cmdline(vb_data);
                 cmdline[cmdlen] = ' ';
-                memcpy(cmdline + cmdlen + 1, vb_cmdline, vb_cmdlen);
```

Comment:
it seem that there's no BUFFER_LEN for the "cmdline". the caller doesn't give us the buffer size to hold the "cmdline". And we will do the operation in "cmdline" memory directly.
I have 2 suggestions here. 1) allocate memory in kernelflinger and use that memory to do operation. 2) point out why it will not lead to any problem (will it overwrite other binaries' memory?). simply change it to memcpy_s() with buffer_size = count cannot help.

Reply:
Isn't it already allocated in line 1150 or 1160?

Reply:
I saw them. then, you should use cmdsize in memcpy_s().
BTW, in line 1143 there's a check for uefi BIOS. I don't think for Celadon we will work with non-efi BIOS. If it is true, you don't need to remove it in this patch. Please record it and we can do it later.

Reply:
Done & recorded